Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java

```diff
@@ -141,18 +139,34 @@ protected void addCompletions(@NotNull final CompletionParameters parameters,
                  }
 
                  updatedResultSet.addElement(lookupElement);
-               }, (includedSet, includedKinds, includedRelevanceTags) -> {
+               }, (includedSet, includedKinds, includedRelevanceTags, libraryFile) -> {
                  final AvailableSuggestionSet suggestionSet = das.getAvailableSuggestionSet(includedSet.getId());
                  if (suggestionSet == null) {
                    return;
                  }
 
+                 final Map<String, Map<String, Set<String>>> existingImports = das.getExistingImports(libraryFile);
                  for (AvailableSuggestion suggestion : suggestionSet.getItems()) {
                    final String kind = suggestion.getElement().getKind();
                    if (!includedKinds.contains(kind)) {
                      continue;
                    }
 
+                   Set<String> declaringLibraries = new HashSet<>();
+                   for (Map.Entry<String, Map<String, Set<String>>> entry : existingImports.entrySet()) {
+                     String importedLibraryUri = entry.getKey();
+                     Map<String, Set<String>> importedLibrary = entry.getValue();
+                     // Checks whether any of the libraries exported by this import declares the same name as this suggestion.
+                     if (importedLibrary.values().stream().anyMatch(names -> names.contains(suggestion.getLabel()))) {
+                       declaringLibraries.add(importedLibraryUri);
```

Comment:
`declaringLibraries` -> `importedLibraries` like "libraries which import provide the suggestion".

Reply:
Done.